### ISSUE TITLE
Define API request in markdown code block

### ISSUE THREAD

**Issue by mAAdhaTTah** [Rooyca/obsidian-api-request#1]
I love the idea of organizing my API requests in Obsidian, but the way the plugin currently works, defining all the info in settings, is a bit limited for that purpose. What would you think about adding a feature so you can define the API request in a markdown code block so the preview would show the result of the API request?

**Comment by adam1279** [comment]
I fully agree. That would be really useful.

**Comment by Rooyca** [comment]
That's a great idea. I would try implementing that after a finish a little project I'm doing now. I'll keep you guys post it.

**Comment by Rooyca** [comment]
I launched a first version (very rudimentary). Just for GET requests with the option to show a specific field from the response. I'll try to keep working on this.

**Comment by mAAdhaTTah** [comment]
@Rooyca I'm also happy to pitch in here if that's helpful.

**Comment by Rooyca** [comment]
> @Rooyca I'm also happy to pitch in here if that's helpful.

That would be great!

**Comment by Rooyca** [comment]
Is anyone still interested on working on this?

**Comment by Bodge-IT** [comment]
I'm interested in using this

**Comment by mAAdhaTTah** [comment]
For some extra context, I took an initial look at this and tried to sketch out what would be required to do what I was hoping to do. I realized the implementation would be quite a bit more complex than I originally anticipated, so I spent some time looking into whether this had been solved in some other form. I found a few projects that were in the same vein:

* [Lama2](https://hexmos.com/lama2/index.html) - Billed explicitly as "Markdown for APIs". Written in Go.
* [Hurl](https://hurl.dev/) - Designed for testing APIs, so more features. Written in Rust.
* [Karate](https://karatelabs.github.io/karate/) - My favorite of the syntaxes (Gherkin-inspired and very readable), but written in Java.
* [vscode-restclient](https://github.com/Huachao/vscode-restclient) - Syntax inspired by RFC2616. Written in JavaScript.

vscode-restclient is probably the easiest to take inspiration and/or crib code from, although seems deeply integrated with VSCode such that it might be difficult to integrate into Obsidian. Of the other 3, Hurl seemed the most feasible to integrate (Rust via wasm is a common workflow, but would need to know/learn Rust), although if we wanted a custom implementation, writing a Karate parser is probably easiest (maybe write a wrapper around a Gherkin parser?).

Obviously, any of these approaches are pretty involved, so if you want something simple, maybe this isn't the best approach, but wanted to share what I found when I looked into this. Unfortunately, I don't have the time to take this further at this time.
